pkg/controller/controller_test.go

```diff
@@ -2838,6 +2840,47 @@ func TestDelete(t *testing.T) {
 			},
 			expectErr: true,
 		},
+		"fail - delete when attached to node": {
+			persistentVolume: &v1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: pvName,
+				},
+				Spec: v1.PersistentVolumeSpec{
+					PersistentVolumeSource: v1.PersistentVolumeSource{
+						CSI: &v1.CSIPersistentVolumeSource{
+							VolumeHandle: "vol-id-1",
+						},
+					},
+					ClaimRef: &v1.ObjectReference{
+						Name: "sc-name",
+					},
+					StorageClassName: "sc-name",
+				},
+			},
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "sc-name",
+				},
+				Parameters: map[string]string{
+					prefixedProvisionerSecretNameKey: "static-${pv.name}-${pvc.namespace}-${pvc.name}",
+				},
+			},
+			volumeAttachment: &storagev1.VolumeAttachment{
```

Comment:
Can you also add a test case where there's a VolumeAttachment that exists for a different PV?

Reply:
done